website/docs/r/app_service_source_control_slot.html.markdown

```diff
@@ -0,0 +1,138 @@
+---
+
+# azurerm_app_service_source_control_slot
+
+Manages an App Service Source Control Slot.
+
+!> **Note:** This Resource is coming in version 3.0 of the Azure Provider and is available **as an opt-in Beta** - more information can be found in [the upcoming version 3.0 of the Azure Provider](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-overview).
+
+## Example Usage
+
+```hcl
+provider "azurerm" {
+	features {}
+}
+
+resource "azurerm_resource_group" "example" {
+	name     = "example-resources"
+	location = "West Europe"
+}
+
+resource "azurerm_service_plan" "example" {
+	name                = "example-plan"
+	resource_group_name = azurerm_resource_group.example.name
+	location            = "West Europe"
+	os_type             = "Linux"
+	sku_name            = "P1V2"
+}
+
+resource "azurerm_linux_web_app" "example" {
+	name                = "example-web-app"
+	resource_group_name = azurerm_resource_group.example.name
+	location            = azurerm_service_plan.example.location
+	service_plan_id     = azurerm_service_plan.example.id
+
+	site_config {}
+}
+
+resource "azurerm_linux_web_app_slot" "example" {
+	name           = "example-slot"
+	app_service_id = azurerm_linux_web_app.example.id
+
+	site_config {}
+}
+
+resource "azurerm_app_service_source_control_slot" "example" {
+	slot_id  = azurerm_linux_web_app_slot.example.id
+	repo_url = "https://github.com/Azure-Samples/python-docs-hello-world"
+	branch   = "master"
+}
+
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `slot_id` - (Required) The ID of the Web App Slot or Function App Slot. Changing this forces a new resource to be created.
+
+---
+
+* `branch` - (Optional) The URL for the repository. Changing this forces a new resource to be created.
+
+* `github_action_configuration` - (Optional) A `github_action_configuration` block as detailed below.
+
+* `repo_url` - (Optional) The branch name to use for deployments. Changing this forces a new resource to be created.
+
+* `rollback_enabled` - (Optional) Should the Deployment Rollback be enabled? Defaults to `false` Changing this forces a new resource to be created.
+
+* `use_local_git` - (Optional) Should the Slot use local Git configuration. Changing this forces a new resource to be created.
+
+* `use_manual_integration` - (Optional) Should code be deployed manually. Set to `true` to disable continuous integration, such as webhooks into online repos such as GitHub. Defaults to `false`. Changing this forces a new resource to be created.
+
+* `use_mercurial` - (Optional) The repository specified is Mercurial. Defaults to `false`. Changing this forces a new resource to be created.
+
+---
+
+A `github_action_configuration` block supports the following:
+
+* `code_configuration` - (Optional) A `code_configuration` block as detailed below.
+
+* `container_configuration` - (Optional) A `container_configuration` block as detailed below.
```

Comment:
Could we add a test for this block?

Reply:
Unfortunately, not meaningfully, no. It requires write access to a GitHub repository to create a GHA / Workflow file(s). It _appears_ to work as far as I could take it manually, but it's not something that can be reasonably done from the AccTests.